### DISCUSSION TITLE
ESP32-8048S070 Board Support (ESP32-S3)

### DISCUSSION THREAD

**Discussion by kzim200** [HASwitchPlate/openHASP#413] [Q&A]
I bought this panel ESP32-S3-pantalla inteligente HMI 8M PSRAM 16M Flash.  It is the one you posted a link to during Dr Z's live stream.

I have tried the varies esp32 flash's listed on the web install page.  Each one says it does not support esp32-S3.  Btw when I try to flash it, it shows the com port and not the name of the panel

**Comment by fvanroie** [comment]
> I have tried the varies esp32 flash's listed on the web install page. Each one says it does not support esp32-S3.

These are brand new devices and we are in the process of adding support for these boards. You can install the development version from https://nightly.openhasp.com/

> Btw when I try to flash it, it shows the com port and not the name of the panel

Yes, this is normal. Please select the correct port to flash the panel. Also check the "Erase Flash" option.

**Comment by kzim200** [comment]
Thank you.  I will give that a try.

**Comment by kzim200** [comment]
Unfortunately, none of the nightlies worked for me.

**Reply by fvanroie** [reply]
You don't have to try them all... just the _one_ specific to your board.
Which board did you buy exactly? (partnumber and/or purchase link please)
What firmware file did you flash to the board?

**Comment by kzim200** [comment]
Here is the link to the board https://es.aliexpress.com/item/1005004952726089.html?spm=a2g0o.order_list.order_list_main.5.21ef194d2vFigv&gatewayAdapt=glo2esp

On the back it has: ESP32-8048S070
and EspressIF   ESP32-S3-WROOM-1

I do not see this listed, so I have tried various ones, the GS-T3E flashes and connects to the wifi, but does not display.  I understand this is most likely an incorrect file.

**Reply by fvanroie** [reply]
You need to flash Full firmware: [sunton-8048s070c_full_16MB_v0.6.4-dev_c52d69b.bin](https://nightly.openhasp.com/firmware/sunton-8048s070c_full_16MB_v0.6.4-dev_c52d69b.bin) from the nightly Web installer.
![image](https://user-images.githubusercontent.com/15969459/210349566-f74d56ad-7b43-42e3-b82f-b1b6fe531198.png)
That one has the specific pin and driver configurations for your board.

**Reply by Tepeinc** [reply]
Hello kzim200!

I think I have the same board, but it's branded Sunton. It looks the same. Link here: https://www.aliexpress.com/item/1005004952726089.html?spm=a2g0o.order_list.order_list_main.65.21ef1802FLUjG4
 I flashed it with Sunton-8048s070c_16MB firmware from nightly build and it is working. It's not working 100% and I think I have a problem in power feed. I think it might need more power than my USB2 or USB3 can deliver because it's disconnecting randomly...  Can you try that Sunton firmware? Maybe fvanroie can confirm this?

**Reply by josh208** [reply]
I have this Sunton board as well and am *stoked* to see added support for it. THANK YOU!!! I was able to flash it, and it's up and running. MQTT is configured and info shows it's connected and publishing, but it's not being discovered in HA. I do have the HA component installed and have different mcu configured successfully with main branch FW. Is it possible the current dev branch is not working with the HA OpenHasp component?

**Reply by Tepeinc** [reply]
I got mine working in HA when I added # to line 12 in /config/custom_components/openhasp/manifest.json. It should look like this "mqtt": ["hasp/discovery/#"]

Still I have some problems... My board is turning off randomly and sometimes touch is not responding. How is your board working? Could I have faulty unit?

**Reply by josh208** [reply]
@Tepeinc That did the trick, thank you! I haven't done much with the Sunton yet, but it's been sitting on my desk since last night and AFAIK it has not rebooted and appears to be working still.

**Reply by elgerg** [reply]
@Tepeinc RE:

> Hello kzim200!
> 
> I think I have the same board, but it's branded Sunton. It looks the same. Link here: https://www.aliexpress.com/item/1005004952726089.html?spm=a2g0o.order_list.order_list_main.65.21ef1802FLUjG4 I flashed it with Sunton-8048s070c_16MB firmware from nightly build and it is working. It's not working 100% and I think I have a problem in power feed. I think it might need more power than my USB2 or USB3 can deliver because it's disconnecting randomly... Can you try that Sunton firmware? Maybe fvanroie can confirm this?

I have one of these, while I'm not using HASP I do get the same sort of issue as mine flickers, I dont get disconnects from what I see but the screen does seem to flicker depending on the viewing angle. Did you manage to sort this out?

Thanks!

**Reply by Tepeinc** [reply]
Hello elgerg!
I still have issuses with my sunton. I use wall plug (3.1A) with it now and the flickering is gone. When I try to go to the OpenHASP's IP page it gets stuck and if I press the esp32 chip with my finger it will work instantly... I think I might try to resolder some of the pads and see if that helps... I ordered 5" version from Sunton, lets see if that works differently.

**Reply by gauthier12** [reply]
Hi.
I have the same flickering issues. I tried using a wall plug (2.4 A) but I still get flickering.
Strangely, it seems that I get flickering only with dark themes. Light themes don't have flickering effect.
Does anyone notice the same thing?

**Reply by WarC0zes** [reply]
Hi,
I just saw your message, indeed when you press esp32 chip, it works instantly.
Would there be a solution?

EDIT:
I found a solution with an aluminum tab. I used cat jelly paper, but aluminum foil should work too.
![sachet cat](https://github.com/HASwitchPlate/openHASP/assets/91471397/28787827-33ba-40c6-98af-1ade973c6d83)

this is how I placed the aluminum tab.
![IMG_20231231_062141](https://github.com/HASwitchPlate/openHASP/assets/91471397/9bb2ab15-7098-4039-8268-3030a5736cec)
![IMG_20231231_062110](https://github.com/HASwitchPlate/openHASP/assets/91471397/29673df3-7cfc-4c6b-9c2d-c2b90609845c)

With just this piece of aluminum, the esp32 works great. I have turned on the ESP several times and it boots directly to OpenHASP and everything works. No latency or loss of WiFi, in my opinion it's part of the chipset that is poorly protected from interference.

![sunton 7inch5](https://github.com/HASwitchPlate/openHASP/assets/91471397/ae9ec035-e86f-4643-8985-c904ce1138b2)

**Reply by Nortonko** [reply]
@WarC0zes can you share your settings please?

**Reply by WarC0zes** [reply]
you can find my config on this page https://forum.hacf.fr/t/openhasp-utilizer-un-esp-avec-ecran-tactile-pour-controler-home-assistant/32713

**Reply by hmorkemo** [reply]
Thanks @WarC0zes and @romainrossi 
I added a piece of copper tape behind the antenna, and the wifi problem seems to have disappeared.
It has been working for several hours without problems now.
I ping it once every 10 seconds. Before the copper tape, it failed randomly, but at least 1 in 4 pings failed. After the tape: No fails!

**Reply by CiroMirkin** [reply]
@WarC0zes  can you share me the model of the frame?

**Reply by WarC0zes** [reply]
> @WarC0zes can you share me the model of the frame?

https://www.printables.com/model/350263-esp32-8048s070-nc-case/comments/688975

**Reply by CiroMirkin** [reply]
@WarC0zes Thank you

**Comment by kzim200** [comment]
SUCCESS!!!!!

Thank you.

I had purposely not used the Sunton.

Thank you again. I am looking forward to seeing what all I can do with
Openhasp.

**Comment by primeroz** [comment]
Just got my Sunton7 from aliexpress and got the board flashed and running.

Do you know if it supports Dimming and what GPIO Pin should i select in the display settings ?

**Reply by fvanroie** [reply]
It indeed supports dimming on `GPIO2`

**Reply by jf1452** [reply]
How do I select GPIO2? It's not available in the list.

**Reply by gauthier12** [reply]
> How do I select GPIO2? It's not available in the list.

I'm asking myself the same question. GPIO2 disappeared when updating from rc4 to rc12

**Comment by WarC0zes** [comment]
Hi,
I bought a Sunton 8048S070C and flashed them to version 0.7.0-rc9. dimming doesn't work, screen never turns off.
With the backlight switch, the screen turns off.
Has anyone tested this version with the Sunton 7inch ?

**Reply by fvanroie** [reply]
afaik The Sunton ESP32-8048S070 does not have a backlight switch... maybe you have another board or a different board revision.

**Reply by WarC0zes** [reply]
Sorry, I didn't specify that I use home assistant, and that the switch comes from the integration.

It's the ESP32-8048S070 model and the revision is v1.3.
![sunton 7inch](https://github.com/HASwitchPlate/openHASP/assets/91471397/68b7d181-e163-46d1-b226-d8654b2515b2)

I have a backlight switch, which turns the screen on and off. The brightness works, I can modify it.
![ha device](https://github.com/HASwitchPlate/openHASP/assets/91471397/7e363f83-b7aa-4173-a9d2-a0440619ec19)

I configured the sleep time.
![idle](https://github.com/HASwitchPlate/openHASP/assets/91471397/6f580da6-ce05-42a6-83f2-e0071044a1a0)
![idle2](https://github.com/HASwitchPlate/openHASP/assets/91471397/f4d9f23c-4cf2-4246-9e91-8d5cf289d947)

The screen doesn't automatically dim or turn off.
I've an Elecrow WZ8048C070 ( similar card, only 4MB flash ) with version 0.7.0-rc8 and it works well, the screen goes to sleep.

**Reply by WarC0zes** [reply]
Hi,
I updated to version 0.7.0-rc9 b7283c9, and the screen goes to sleep. Problem solved.

**Reply by pepe59** [reply]
Yes, the version you are using works.
Attention to version v0.7.0-rc9_33d21e0.
Here the screen off control does not work.

**Reply by ArtSoft66** [reply]
The screen off control also does not work on the Sunton 8048S050C board.

**Reply by bossmaster69000** [reply]
help plz i dont know what im doing

**Reply by fvanroie** [reply]
I updated https://nightly.openhasp.com/ to r0.7.0-rc10, that should fix the backlight dimming bug.

**Reply by pepe59** [reply]
Yes, I can confirm that the Sunton 8048S050C display with
fw r0.7.0-rc10 works fine dimming.

**Reply by gauthier12** [reply]
Hi.
I've tried to update to rc12 in  https://nightly.openhasp.com/ but backlight dimming does not work.
I used this file : 
https://nightly.openhasp.com/firmware/sunton-8048s070c_full_16MB_v0.7.0-rc12_cd34d89.bin
In the display setting section, GPIO02 is not available
![image](https://github.com/user-attachments/assets/bc57dd23-8463-4be5-82bb-3d6ee0a6a531)
Is there a way to find an old build to test ?
How were you able to configure backlight ?
I've tried to set GPIO02 in the json config file but no success
Cheers

**Reply by gauthier12** [reply]
Ok. I've found that the backlight control is working when sending the MQTT command `{"state":"on","brightness":12}` on the topic ` hasp/plate/command/backlight`
I can check that the plate is going to idle but the screen isn't going off by itself.
I don't understand why the screen isn't going off by itself when idle

**Reply by gauthier12** [reply]
Ok, I figured it out (and I'm talking alone). I had to create 3 scripts idle_off.cmd, idle_short.cmd and idle_long.cmd
Inside this script, configure the desired backlight.
I didn't find this system in the documentation, 
Sorry for the spam

**Comment by nicolo09** [comment]
Hi, I bought this panel and i'm trying to add an external UART device connected to the top connector labeled P4, it should be connected to UART1 but when i try to use Serial1 in my code the screen suddenly shift to a greenish color and the serial can't connect to the sensor. Does anyone knows if UART1 is connected to something else except for the JST header? If so where can I connect my sensor?
